daemon/logger/jsonfilelog/read.go

```diff
@@ -174,6 +175,15 @@ func followLogs(f *os.File, logWatcher *logger.LogWatcher, notifyRotate chan int
 		fileWatcher.Close()
 	}()
 
+	logWatcherClosed := make(chan struct{})
+	go func() {
```

Comment:
Good shout! I've just pushed an update to the code, does it look a little better? If it's not quite how you imagined let me know.